blackbox/env/capabilities.py

```
@@ -26,6 +26,47 @@
+            line.strip() == "user_allow_other" or line.strip().startswith("user_allow_other ")
+            for line in text.splitlines()
+            if not line.strip().startswith("#")
+        )
+        if has_directive:
+            return
+
+    sudo = shutil.which("sudo")
+    if sudo is None:
+        return
+    # Probe passwordless sudo before attempting any write.
+    probe = subprocess.run([sudo, "-n", "true"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=False)
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)